.test-infra/jenkins/job_ClusterCredentialsRotation.groovy

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import CommonJobProperties as commonJobProperties
+
+job('Rotate Cluster Credentials') {
+  description('Rotates Certificates and performs an IP rotation for metrics and io-datastores')
+
+  // Set common parameters.
+  commonJobProperties.setTopLevelMainJobProperties(delegate)
+
+  // Sets that this is a cron job.
+  commonJobProperties.setCronJob(delegate, 'H 2 1 */2 *')// At 00:02am every second month.
+
+  steps {
+
+    //Credentials rotation for metrics and io-datastores
+
+    //Set a maintenance window
+    shell('''gcloud container clusters update metrics \
+    --zone=us-central1-a --maintenance-window=06:00 --quiet''')
+
+    shell('''gcloud container clusters update io-datastores \
+    --zone=us-central1-a --maintenance-window=06:00 --quiet''')
+
+    //Starting credential rotation
+    // it's necessary to rebuild the nodes after rotation to avoid apiservices issues
+    shell('''gcloud container clusters update metrics \
+    --start-credential-rotation --zone=us-central1-a --quiet''')
+
+    shell('''gcloud container clusters update io-datastores \
+    --start-credential-rotation --zone=us-central1-a --quiet''')
+
+    //Rebuilding the nodes
+    shell('''gcloud container clusters upgrade metrics \
```

Comment:
What happens if these commands close with an error state?

Reply:
If anything goes wrong after line 50, GCP will automatically complete the process after 7 days ([Automatic completion](https://cloud.google.com/kubernetes-engine/docs/how-to/credential-rotation#performing_a_credential_rotation)). 

If something fails during the first 2 commands (_--start-credential-rotation_), the rotation will not start and the cluster will keep the previous credentials. We will add a condition to only execute the following steps if the start-credential-rotation part was successful. I could also add an email alert in case of failure.

Reply:
Please add the condition and email alert. I also recommend making each cluster independent, with separate alerts in case of failure. So, try to start rotation for io-datastores, complete upgrade and update, then do the same for metrics.

Reply:
Sure, I checked Jenkins configuration and it has the -**xe** flag when execution DSL Jobs, so if a command returns an error the script stops and it will be marked as a fail, I separated the rotation with one job per cluster with separated email notifications to _dev_,  pointing to JOB_URL and JENKINS_URL for further details.